server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -579,7 +579,8 @@ else if (!addFuture) {
         if (poolIds != null && !poolIds.isEmpty()) {
             criteria = this.currentSession()
                 .createCriteria(Pool.class)
-                .add(CPRestrictions.in("id", poolIds));
+                .createAlias("product", "Product");
+                criteria.add(CPRestrictions.in("id", poolIds));
```

Comment:
This should be unindented, or we can drop the the criteria reference and move the .add to above the creation of the product alias.